# PANDAS DATA CORRELATIONS

Finding realtionships-
The great aspect of Pandas module is corr() method. The corr() method calculates the relationship between each column in your data set.



In [1]:
import pandas as pd

In [2]:
df = pd.read_csv('data.csv')

In [ ]:
df.corr() # the corr() method ignores 'not numeric methods'

,Duration,Pulse,Maxpulse,Calories
Duration,1.000000,-0.155408,0.009403,0.922717
Pulse,-0.155408,1.000000,0.786535,0.025121
Maxpulse,0.009403,0.786535,1.000000,0.203813
Calories,0.922717,0.025121,0.203813,1.000000


The result of the `corr()` method is a table with a lot of numbers that represents how well the relationship is between two columns.

The number varies from -1 to 1

The 1 means that there is a relationship(a perfect correlation), and for this data set, each time a value went up to in the first column, the other one went up as well

0.9 is a also a very good relationship, and if you increase one value the other will probably increase as well

-0.9 would be just as good relationship as 0.9, but if you increase one value, the other will probably go down.

0.2 means NOT a good realationship, meaning that if one valus goes up does not mean that the other will.



__Perfect Correlation__
We can see that "Duration" and "Duration" got the number `1.000000`, which makes sense, each column always has a perfect relationship with itself.

__Good Correlation__
"Duration" and "Calories" got a `0.922721` correlation, which is a very good correlation, and we can predict that the longer you work out, the more calories you burn, and the other way around: if you burned a lot of calories, you probably had a long work out.

__Bad Correlation__ 
"Duration" and "Maxpulse" got a `0.009403` correlation, which is a very bad correlation, meaning that we can not predict the max pulse by just looking at the duration of the work out, and vice versa.

